src/models/linear_regression_benchmark/federated/train.py

```diff
@@ -0,0 +1,131 @@
+        public_features = load_public_airport_features(airline_labels.index, airport)
+        private_features = load_private_airline_airport_features(
+            airline_labels.index, airline, airport
+        )
+
+        airline_features = pd.concat([private_features, public_features], axis=1)
+
+        features.append(airline_features)
+        labels.append(airline_labels)
+
+    features = pd.concat(features, axis=0)
+    labels = pd.concat(labels, axis=0)
+
+    model = sio.load("linear_regression_benchmark.skops", trusted=True)
+    set_initial_params(model)
+
+    @dataclass
+    class LinearRegressionTrainClient(fl.client.NumPyClient):
+        """Linear regression flower client"""
+
+        airline: str
+
+        def get_parameters(self, config):
+            return get_model_parameters(model)
+
+        def fit(self, parameters, config):
+            logger.debug(f"Fitting client {self.airline}")
+            set_model_params(model, parameters)
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                model.fit(features, labels)
+                logger.info(
+                    f"""{self.airline} Training finished for round {config["server_round"]}"""
+                )
+
+            return get_model_parameters(model), len(labels), {}
+
+        def evaluate(self, parameters, config):
+            logger.debug(f"Evaluating client {self.airline}")
+            set_model_params(model, parameters)
+            predictions = model.predict(features)
+            loss = mean_absolute_error(labels, predictions)
+            return loss, len(labels), {}
+
+    return LinearRegressionTrainClient(airline)
+
+
+def fit_config(server_round: int) -> Dict:
+    """Send round number to client."""
+    return {"server_round": server_round}
+
+
+class SaveModelStrategy(fl.server.strategy.FedAvg):
+    def aggregate_fit(self, server_round, results, failures):
+        aggregated_parameters, aggregated_metrics = super().aggregate_fit(
+            server_round, results, failures
+        )
+
+        if aggregated_parameters is not None:
+            aggregated_ndarrays: List[np.ndarray] = fl.common.parameters_to_ndarrays(
+                aggregated_parameters
+            )
+
+            logger.info(f"Saving round {server_round} aggregated_ndarrays")
+            np.savez("weights.npz", *aggregated_ndarrays)
+
+        return aggregated_parameters, aggregated_metrics
+
+
+if __name__ == "__main__":
+    client_resources = {
+        "num_cpus": int(os.cpu_count() / 2) + 1,
+    }
+
+    model = sio.load("linear_regression_benchmark.skops", trusted=True)
+    set_initial_params(model)
+
+    strategy = SaveModelStrategy(
+        min_available_clients=len(AIRLINES),
+        on_fit_config_fn=fit_config,
+    )
+
+    fl.simulation.start_simulation(
+        client_fn=train_client_factory,
+        client_resources=client_resources,
+        num_clients=len(AIRLINES),
```

Comment:
Could use `client_ids=AIRLINES` instead, so the IDs are directly the name of the airline instead of a stringified number. But this way works just as well!